OpenOversight/app/email_client.py

```diff
@@ -1,56 +1,160 @@
-from apiclient import errors
+import os.path
+from abc import ABC, abstractmethod
+from threading import Thread
+from typing import Optional
+
 from flask import current_app
+from flask_mail import Mail, Message
 from google.oauth2 import service_account
 from googleapiclient.discovery import build
 
 from OpenOversight.app.models.emails import Email
 from OpenOversight.app.utils.constants import KEY_OO_SERVICE_EMAIL, SERVICE_ACCOUNT_FILE
 
 
-class EmailClient(object):
+class EmailProvider(ABC):
+    """Base class to define how emails are sent."""
+
+    @abstractmethod
+    def start(self):
+        """Set up the email provider."""
+
+    @abstractmethod
+    def is_configured(self):
```

Comment:
Could you mark what the outputs are? ` def is_configured(self: EmailProvider) -> bool:`

Reply:
I've annotated `is_configured` with the `bool` return type. Let me know if there's anything else that needs to be annotated!

Reply:
Did you forget to push that change? I'm not seeing it in the code.

Reply:
Sorry for the confusion! I was about to push my changes but had a couple other small things I wanted to include